apps/speed.c

```diff
@@ -3161,6 +3566,369 @@ int speed_main(int argc, char **argv)
+                BIO_printf(bio_err, "Encaps error for %s.\n", kem_name);
+                goto kem_err_break;
+            }
+            /* Now prepare decaps data structs */
+            kem_decaps_ctx = EVP_PKEY_CTX_new_from_pkey(app_get0_libctx(),
+                                                        pkey,
+                                                        app_get0_propq());
+            if ((kem_decaps_ctx == NULL) ||
+                (EVP_PKEY_decapsulate_init(kem_decaps_ctx, NULL) <= 0)  ||
+                ((!strncmp(kem_name, "RSA", 3) &&
+                 (EVP_PKEY_CTX_set_kem_op(kem_decaps_ctx, "RSASVE") <= 0))) ||
+                (((!strncmp(kem_name, "EC", 2)) ||
+                  (!strcmp(kem_name, "X25519")) || 
+                  (!strcmp(kem_name, "X448"))) && 
+                 (EVP_PKEY_CTX_set_kem_op(kem_decaps_ctx, "DHKEM") <= 0)) ||
+                (EVP_PKEY_decapsulate(kem_decaps_ctx, NULL, &rcv_secret_len,
+                                      out, out_len) <= 0)) {
+                BIO_printf(bio_err,
+                           "Error while initializing decaps data structs for %s.\n",
+                           kem_name);
+                goto kem_err_break;
+            }
+            rcv_secret = OPENSSL_malloc(rcv_secret_len);
+            if (rcv_secret == NULL) {
+                BIO_printf(bio_err, "MemAlloc failure in decaps for %s.\n",
+                           kem_name);
+                goto kem_err_break;
+            }
+            if ((EVP_PKEY_decapsulate(kem_decaps_ctx, rcv_secret,
+                                      &rcv_secret_len, out, out_len) <= 0) ||
+                (rcv_secret_len != send_secret_len) ||
+                (memcmp(send_secret, rcv_secret, send_secret_len))) {
+                BIO_printf(bio_err, "Decaps error for %s.\n", kem_name);
+                goto kem_err_break;
+            }
+            loopargs[i].kem_gen_ctx[testnum] = kem_gen_ctx;
+            loopargs[i].kem_encaps_ctx[testnum] = kem_encaps_ctx;
+            loopargs[i].kem_decaps_ctx[testnum] = kem_decaps_ctx;
+            loopargs[i].kem_out_len[testnum] = out_len;
+            loopargs[i].kem_secret_len[testnum] = send_secret_len;
+            loopargs[i].kem_out[testnum] = out;
+            loopargs[i].kem_send_secret[testnum] = send_secret;
+            loopargs[i].kem_rcv_secret[testnum] = rcv_secret;
+            break;
+
+            kem_err_break:
+                ERR_print_errors(bio_err);
+                op_count = 1;
+                kem_checks = 0;
+                break;
+
+        }
+        if (kem_checks != 0) {
+            kskey_print_message(kem_name, "keygen", kems_c[testnum][0],
+                                 seconds.kem);
+            Time_F(START);
+            count =
+                run_benchmark(async_jobs, KEM_keygen_loop, loopargs);
+            d = Time_F(STOP);
+            BIO_printf(bio_err,
+                       mr ? "+R12:%ld:%s:%.2f\n" :
```

Comment:
`R12 ` seems to be already used, no ?

Reply:
correct. Numbering updated in https://github.com/openssl/openssl/pull/19968/commits/b7b2a70ec374b2d556e1290a0abceb48009ae939